helix-core/src/selection.rs

```diff
@@ -122,6 +122,15 @@ impl Range {
         }
     }
 
+    // flips the direction of the selection
+    pub fn flip(&self) -> Self {
```

Comment:
Range is `Copy` so `self` can be used:

```suggestion
    pub fn flip(self) -> Self {
```

Doesn't seem to change the compiler output though. I also attempted to use 
```rust
    pub fn flip(mut self) -> Self {
        std::mem::swap(&mut self.head, &mut self.anchor);
        self
    }
```
to see if it would generate better assembly but it ended up worse: https://rust.godbolt.org/z/rP8vva9G6